File: python/2917.py
from typing import List

class Solution:
    def findKOr(self, nums: List[int], k: int) -> int:
        ans = 0
        for i in range(31):
            cnt = sum(num >> i & 1 for num in nums)
            if cnt >= k:
                ans |= 1 << i
        return ans

File: go/2917.go
package main

func findKOr(nums []int, k int) (res int) {
	bitC := make([]int, 32)
	var bitCounts func(num int)
	bitCounts = func(num int) {
		for i := 0; num > 0; i++ {
			if num&1 > 0 {
				bitC[i]++
			}
			num >>= 1
		}
	}
	for _, num := range nums {
		bitCounts(num)
	}
	t := 1
	for _, c := range bitC {
		if c >= k {
			res += t
		}
		t <<= 1
	}
	return
}
